pytorch_lightning/trainer/training_loop.py

```diff
@@ -643,15 +642,15 @@ def optimizer_closure():
                     self.batch_loss_value = 0
                     self.avg_loss = np.mean(self.running_loss[-100:])
 
-        # activate batch end hook
-        if self.is_function_implemented('on_batch_end'):
-            model = self.get_model()
-            with self.profiler.profile('on_batch_end'):
+        # Batch end events
+        with self.profiler.profile('on_batch_end'):
+            # callbacks
+            self.on_batch_end()
+            # model hooks
+            if self.is_function_implemented('on_batch_end'):
+                model = self.get_model()
```

Comment:
I would consider replacing all these twp lineners so there is no need if the model is needed later...
`self.get_model().on_...()`